schemavalidator/conflicts_with_test.go

```diff
@@ -0,0 +1,292 @@
+		req       tfsdk.ValidateAttributeRequest
+		in        path.Expressions
+		expErrors int
+	}
+
+	testCases := map[string]testCase{
+		"base": {
+			req: tfsdk.ValidateAttributeRequest{
+				AttributeConfig:         types.String{Value: "bar value"},
+				AttributePath:           path.Root("bar"),
+				AttributePathExpression: path.MatchRoot("bar"),
+				Config: tfsdk.Config{
+					Schema: tfsdk.Schema{
+						Attributes: map[string]tfsdk.Attribute{
+							"foo": {
+								Type: types.Int64Type,
+							},
+							"bar": {
+								Type: types.StringType,
+							},
+							"baz": {
+								Type: types.Int64Type,
+							},
+						},
+					},
+					Raw: tftypes.NewValue(tftypes.Object{
+						AttributeTypes: map[string]tftypes.Type{
+							"foo": tftypes.Number,
+							"bar": tftypes.String,
+							"baz": tftypes.Number,
+						},
+					}, map[string]tftypes.Value{
+						"foo": tftypes.NewValue(tftypes.Number, 42),
+						"bar": tftypes.NewValue(tftypes.String, "bar value"),
+						"baz": tftypes.NewValue(tftypes.Number, 43),
+					}),
+				},
+			},
+			in: path.Expressions{
+				path.MatchRoot("foo"),
+				path.MatchRoot("baz"),
+			},
+			expErrors: 2,
+		},
+		"conflicting-is-nil": {
+			req: tfsdk.ValidateAttributeRequest{
+				AttributeConfig:         types.String{Value: "bar value"},
+				AttributePath:           path.Root("bar"),
+				AttributePathExpression: path.MatchRoot("bar"),
+				Config: tfsdk.Config{
+					Schema: tfsdk.Schema{
+						Attributes: map[string]tfsdk.Attribute{
+							"foo": {
+								Type: types.Int64Type,
+							},
+							"bar": {
+								Type: types.StringType,
+							},
+						},
+					},
+					Raw: tftypes.NewValue(tftypes.Object{
+						AttributeTypes: map[string]tftypes.Type{
+							"foo": tftypes.Number,
+							"bar": tftypes.String,
+						},
+					}, map[string]tftypes.Value{
+						"foo": tftypes.NewValue(tftypes.Number, nil),
+						"bar": tftypes.NewValue(tftypes.String, "bar value"),
+					}),
+				},
+			},
+			in: path.Expressions{
+				path.MatchRoot("foo"),
+			},
+		},
+		"error_conflicting-is-unknown": {
+			req: tfsdk.ValidateAttributeRequest{
+				AttributeConfig:         types.String{Value: "bar value"},
+				AttributePath:           path.Root("bar"),
+				AttributePathExpression: path.MatchRoot("bar"),
+				Config: tfsdk.Config{
+					Schema: tfsdk.Schema{
+						Attributes: map[string]tfsdk.Attribute{
+							"foo": {
+								Type: types.Int64Type,
+							},
+							"bar": {
+								Type: types.StringType,
+							},
+						},
+					},
+					Raw: tftypes.NewValue(tftypes.Object{
+						AttributeTypes: map[string]tftypes.Type{
+							"foo": tftypes.Number,
+							"bar": tftypes.String,
+						},
+					}, map[string]tftypes.Value{
+						"foo": tftypes.NewValue(tftypes.Number, tftypes.UnknownValue),
+						"bar": tftypes.NewValue(tftypes.String, "bar value"),
+					}),
+				},
+			},
+			in: path.Expressions{
+				path.MatchRoot("foo"),
+			},
+			expErrors: 1,
```

Comment:
As mentioned above, we may need to decide more carefully on the semantics with unknowns, or let provider developers choose between `ConflictsWith` variants (although I worry there's ultimately a false hope that handling unknowns as failing would drive everyone towards using the one that doesn't fail on unknowns due to practitioner complaints).

Reply:
Yeah, I think we should really try to be lenient with `Unknowns`: better to NOT fail the validation and only do it once all the involved attributes are known.